src/Extension/CoreExtension.php

```diff
@@ -543,13 +543,22 @@ function twig_date_converter(Environment $env, $date = null, $timezone = null)
  *
  * @return string
  */
-function twig_replace_filter($str, $from)
+function twig_replace_filter($str, $from, $limit = null)
 {
     if (!twig_test_iterable($from)) {
         throw new RuntimeError(sprintf('The "replace" filter expects an array or "Traversable" as replace values, got "%s".', \is_object($from) ? \get_class($from) : \gettype($from)));
     }
 
-    return strtr($str, twig_to_array($from));
```

Comment:
For performance the function `strtr` should be used when `$limit === null`.
`array_walk` is know to be relatively slow. https://medium.com/@steveclifton_12558/array-walk-performance-test-b8cfb11511a8